draft-ietf-quic-transport.md

```diff
@@ -1834,26 +1834,32 @@ congestion in the network by setting a codepoint in the IP header of a packet
 instead of dropping it.  Endpoints react to congestion by reducing their sending
 rate in response, as described in {{QUIC-RECOVERY}}.
 
-To use ECN, QUIC endpoints first determine whether a path and peer support ECN
-marking. Verifying the path occurs at the beginning of a connection and when the
-connection migrates to a new path (see {{migration}}).
-
-Each endpoint independently verifies and enables ECN for the path from it to the
-peer.
-
-To verify that both a path and the peer support ECN, an endpoint MUST set one of
-the ECN Capable Transport (ECT) codepoints -- ECT(0) or ECT(1) -- in the IP
-header {{!RFC8311}} of all outgoing packets.
+To use ECN, QUIC endpoints first determine whether a path supports ECN
+marking and the peer is able to access the ECN codepoint in the IP header. 
+A network path regarded as not supporting ECN if ECN marked packets get dropped
+or ECN marking are rewritten on the path.
+An endpoint verifies the path, both during connection establishment 
+and when migrating to a new path (see {{migration}}).
+
+Each endpoint independently verifies and enables use of ECN by setting the IP
+header ECN codepoint to ECN Capable Transport (ECT) for the path from it to the
+other peer. Even if ECN is not used on the path to the peer, the endpoint MUST
+provide ECN feedback information if the receive of a non-zero ECN IP codepoint
+is observed.
```

Comment:
Grammar here could be improved: "an endpoint MUST provide ECN feedback information if it receives packets with non-zero ECN markings." ?

Reply:
Maybe just "the endpoint MUST provide feedback about ECN markings received (if accessible)." Is that better?

Reply:
Yeah, concise and correct.  You can just drop the parenthetical: if the markings aren't accessible, the endpoint didn't get them.